### ISSUE TITLE
[StarBook Mk VI - ITE] Charging stops intermittently

### ISSUE THREAD

**Issue by david-mohr** [StarLabsLtd/firmware#95]
After the latest update, having followed along with the other charging issues (#89, #90), a new issue has appeared:

- laptop is powered on, running on battery
- connect power (DC jack) and external monitor (usb-c, also with charging)
- battery starts charging
- eventually battery is full and the power appears to disconnect so that I'm running on battery
- battery slowly discharges until I notice
- unplug the DC jack and plug it back in and it starts charging again

**Comment by david-mohr** [comment]
100% and the LED is unlit

**Comment by Sean-StarLabs** [comment]
Sounds right - it shouldn't start charging again until its dropped below 95%

**Comment by Sean-StarLabs** [comment]
Okay - it might not be able the two chargers - can I ask why you have two connected?

**Comment by david-mohr** [comment]
It's not intentional... the USB-c is just for my monitor (and its usb hub). If you think it's sufficient to just use that as the charger, then I can remove the DC jack and see how that goes.

**Comment by Sean-StarLabs** [comment]
Depends on the monitor, but worth a try

**Comment by david-mohr** [comment]
It's a dell S2718D. It worked for a while, but eventually it disconnected the power *and* the display, and then went into a tight loop connecting and disconnecting the monitor. Now when I connect the monitor, it will accept power, but not recognise the monitor (monitor says there's no signal). I'll try switching over to DC jack and HDMI instead.

**Comment by cichy1173** [comment]
> It's a dell S2718D. It worked for a while, but eventually it disconnected the power _and_ the display, and then went into a tight loop connecting and disconnecting the monitor. Now when I connect the monitor, it will accept power, but not recognise the monitor (monitor says there's no signal). I'll try switching over to DC jack and HDMI instead.

Do you use Star Labs original charger?

**Comment by david-mohr** [comment]
@cichy1173 yes

**Comment by cichy1173** [comment]
> @cichy1173 yes

Please, try different charger. I had almost the same problem with original charger and USB-C Hub. Laptop charged 100%, and it started disconnecting and connecting. I changed charger, and everything is fine. I think, I need to report this charger issue. I think the charger may have broken.

**Comment by david-mohr** [comment]
Interesting! I've got the same issue now that I'm not using the usb-c connection to the monitor, gets to 100% and will continually disconnect/connect: not dipping to 95%, LED is pink/purple while "charging". Only lasts a couple of seconds then it disconnects again.

**Comment by Sean-StarLabs** [comment]
> Do you use Star Labs original charger?

Where does the charger factor in? If it's a PD monitor, it should have its own supply?

**Comment by cichy1173** [comment]
> > Do you use Star Labs original charger?
> 
> Where does the charger factor in? If it's a PD monitor, it should have its own supply?

No. The monitor does not have a PD. I connect the monitor to a usb-c hub together with a usb-c charger (I connect the charger to this hub) and then the hub to the laptop. 

I have tested the charger together with other computers (and without using the usb-c hub) and there too the charger behaves strangely - for example the keyboard on the Huawei Matebook x Pro started flashing like broken. As with the Starbook, changing the charger helped here too. I wonder if the charger from the starbook kit might not be defective.

**Comment by Sean-StarLabs** [comment]
The Dell S2718D used for this issue does - your issue sounds completely different. Can you open a separate issue with the model of the Hub?

**Comment by david-mohr** [comment]
Just to clarify my situation:

- PD monitor only: power got to to 100% and then the power and display were both disconnected, then both the laptop screen and external screen started to flicker like it was being repeatedly plugged in/unplugged again
- DC jack and HDMI: gets to 100% and will continually disconnect/connect the power every few seconds

**Comment by Sean-StarLabs** [comment]
Can you try https://github.com/StarLabsLtd/firmware/tree/master/StarBook/MkVI-Intel/ite/1.06 with coreboot 8.34?

**Comment by david-mohr** [comment]
sure thing. I haven't done an EFI update before, can I please confirm that these are the instructions that I need to follow? https://starlabsltd.github.io/firmware/methods/efi_shell/

**Comment by Sean-StarLabs** [comment]
Yup

**Comment by david-mohr** [comment]
thanks, I'll jump on it first thing in the morning

**Comment by EverlastingOS** [comment]
How do i update from EFI, have downloaded all files, put it on a fat32 USB in folder "Firmware", boot in EFI shell, and nothing happens.

**Comment by Sean-StarLabs** [comment]
Dont put it in the folder - it wont be able to find it

**Comment by cichy1173** [comment]
> How do i update from EFI, have downloaded all files, put it on a fat32 USB in folder "Firmware", boot in EFI shell, and nothing happens.

![obraz](https://user-images.githubusercontent.com/32648503/228811469-7a21f14f-f904-4c7f-89aa-64602caa411d.png)
I think `Firmware` is a name of memory stick, not folder.

**Comment by EverlastingOS** [comment]
> Dont put it in the folder - it wont be able to find it

Oh, okey, thanks, and do i need download all files from https://github.com/StarLabsLtd/firmware/tree/master/StarBook/MkVI-Intel/ite/1.06

**Comment by Sean-StarLabs** [comment]
Just the zip file

**Comment by EverlastingOS** [comment]
> Dont put it in the folder - it wont be able to find it

Oh, okey, thanks, and do i need download all files from https://github.com/StarLabsLtd/firmware/tree/master/StarBook/MkVI-Intel/ite/1.06

> Just the zip file

Nothing happens, and can not unzip the file.

**Comment by EverlastingOS** [comment]
> > Nothing happens, and can not unzip the file.
> 
> You might be better of waiting for the stable release that'll come via the LVFS, but if you really want to install this testing update, this'll download and extract it:
> 
> ```
> wget https://raw.githubusercontent.com/StarLabsLtd/firmware/master/StarBook/MkVI-Intel/ite/1.06/efi-B6-I.zip
> unzip efi-B6-I.zip 
> ```

Okey, thats why. Can wait for the stable one :)

**Comment by david-mohr** [comment]
I seem to have a problem now that says "the laptop is plugged in, but still discharging" (DC jack only, no external monitor). Then after a short time, the power will disconnect and reconnect, but the error persists and the battery is still discharging.

Down to 8% battery, going to roll back to 1.05

**Comment by david-mohr** [comment]
No change, now sitting at 0% according to the OS. Saw there is now 1.07, will try that

**Comment by david-mohr** [comment]
I seem to be unable to install 1.07 (or I'm checking the installed version the wrong way). I have wiped the usb stick to ensure I have only files from the latest zip. Then I reboot into the EFI shell with the usb attached (although I have found that I need to unplug the usb stick while in the boot menu and plug it in again otherwise the EFI shell with detect the auto boot script, but not execute it).

Install seems to go fine and I power off the machine after it reboots to ensure I don't trigger the "overflow" mentioned in other tickets. I then power the machine on, log in and run `fwupdmgr get-device`, but it reports the embedded controller is still version 1.05. I assume I'm doing something wrong, I just can't figure out what it is.

**Comment by david-mohr** [comment]
As a side note, I've changed the power profile to "power saving" to see if that will help kick start the recharging.

**Comment by david-mohr** [comment]
An interesting development... I was moving my laptop to another room and it powered off (no battery left), but now it seems to be charging after powering it back on and booting up.

**Comment by divico** [comment]
I'm on 8.34/1.05 and I'm suddenly experiencing the same symptoms as described in #89 which are close those described here.

With only the DC jack plugged-in the LED is red and the battery won't charge. The gnome indicator is oscillating between charging and discharging.

**Comment by divico** [comment]
> I'm on 8.34/1.05 and I'm suddenly experiencing the same symptoms as described in #89 which are close those described here.
> 
> With only the DC jack plugged-in the LED is red and the battery won't charge. The gnome indicator is oscillating between charging and discharging.
> 
> I'll try to update as soon as I can

Ok in the meantime I could try 1.06 and 1.07. Same issue with both versions. I have now 2% battery left and no options left to charge my laptop, which unfortunately is now unusable.  Please let me know if I can assist somehow fixing this ASAP. thx!

BTW `although I have found that I need to unplug the usb stick while in the boot menu and plug it in again otherwise the EFI shell with detect the auto boot script, but not execute it).` I can confirm that. If one follows the official instruction it will never update. This important detail should be added to the official doc or fixed if it is not supposed to work like this.

**Comment by cichy1173** [comment]
> Is switching to AMI a workaround for all these charging issue?

You fan downgrade to EC 1.01

**Comment by Sean-StarLabs** [comment]
> I seem to be unable to install 1.07 (or I'm checking the installed version the wrong way). I have wiped the usb stick to ensure I have only files from the latest zip. Then I reboot into the EFI shell with the usb attached (although I have found that I need to unplug the usb stick while in the boot menu and plug it in again otherwise the EFI shell with detect the auto boot script, but not execute it).

`sudo nvramtool -w mirror_flag=Disable` will stop it doing that.

> BTW although I have found that I need to unplug the usb stick while in the boot menu and plug it in again otherwise the EFI shell with detect the auto boot script, but not execute it).

Sorry, don't follow - it's a paradox if it detects the script but doesn't execute it as they are one and the same. It's most likely the USB; the EFI shell is stuff is about 15 years old - no-one changes it because it works! Step by step, what happens when you follow the instructions?

> Ok in the meantime I could try 1.06 and 1.07. Same issue with both versions.

The issue that @david-mohr mentioned is categorically fixed in `1.07` - whatever you experienced was different so you can you let me know:
* EC version
* BIOS version
* Charge level
* LED colours
* Charging cable
* If possible, the output of `ectool -d`

**Comment by divico** [comment]
> Sorry, don't follow - it's a paradox if it detects the script but doesn't execute it as they are one and the same. It's most likely the USB; the EFI shell is stuff is about 15 years old - no-one changes it because it works! Step by step, what happens when you follow the instructions?

In my case I cannot confirm that it does detect the script. Actually I think it doesn't detect the script. In all the cases the issue is that it won't update if the USB stick is inserted before the laptop is turned on. The only procedure that works for me is to turn on the machine with the USB stick disconnected, enter boot manager and insert the USB stick while in the boot manager. Then start the EFI shell and it works. I have a USB stick with an activity LED and I can clearly see that there is no activity on the USB if it is connected before turning on. If it is connected after entering the boot manager, I can clearly see that the activity LED is blinking immediately after plug-in the stick. It is of course not a big issue but if you don't know and you strictly follow the [documentation which says](https://starlabsltd.github.io/firmware/methods/efi_shell/)

`Your USB Flash Drive is now ready. Leave it connected, connect your laptop charger and then turn off your laptop. Turn your laptop back on and boot to the EFI Shell...` 

then it will not work. At least in my (our?) case.

**Comment by divico** [comment]
> The issue that @david-mohr mentioned is categorically fixed in `1.07` - whatever you experienced was different so you can you let me know:
 
Ok thanks for the clarification.

* **EC version** the issue appeared after running fine for a week or so on 1.05. Once the issue appeared, I tried 1.06 and 1.07 which didn't solve. Downgrading to 1.01 immediately solved the issue.
* **BIOS version** coreboot 8.34
* **Charge level** I noticed the issue at 20% because I got the warning from gnome. Most likely it was discharging before and I didn't notice. I went down to 2% before trying the other EC versions mentioned above.
* **LED colours** charging LED would be solid red when a charger is connected, otherwise blinking red. (at least when it was at 2%, >20% I did not pay attention). The activity LED was solid blue.
 * **Charging cable** Starlarbs charger and USB-C / DC-jack cable and nothing else connected when the battery was <20% (i.e. when I realised it was not charching). Before I was on the Starlabs USB-C 12 port hub with PD charger from Trust.com
* **If possible, the output of** `ectool -d`  Ok I'll provide that, need to install first. I guess you need that while the issue occurs? At the moment I'm running on 1.01 and I have no issue. I guess I should first reproduce the issue and then run the dump?

**Comment by Sean-StarLabs** [comment]
> BIOS version coreboot 8.34
`8.34` had the mirror flag forced (coreboot updates the EC) - did you check the EC version after installing it?

**Comment by Sean-StarLabs** [comment]
> Then start the EFI shell, and it works. I have a USB stick with an activity LED, and I can clearly see that there is no activity on the USB if it is connected before turning on. 

Which port? And when you follow the instructions (i.e. have it connected before turning on), does the USB appear int he EFI Shell?

**Comment by david-mohr** [comment]
Just a quick update on my testing of 1.07:

- Running on USB-C power, the battery fully charged and stayed at 100% without any disconnections
- Running on DC-power, it will still disconnect/reconnect once at full power (LED is pink/purple)

Tomorrow I will test with both connected.

**Comment by aaron-gh** [comment]
Can confirm the charger will connect and disconnect when using coreboot if the battery is at the max charge level or close. Not sure about this with USB C but with dc jack charging this seems to be an issue.

**Comment by Sean-StarLabs** [comment]
> will connect and disconnect 

What are you counting as a disconnect?

**Comment by Sean-StarLabs** [comment]
Ah, it finally happened to me - it's LEARN running awry, which explains why it works for days and then doesn't. Fixed that in `1.10`.

**Comment by divico** [comment]
> > BIOS version coreboot 8.34
> 
> `8.34` had the mirror flag forced (coreboot updates the EC) - did you check the EC version after installing it? Just added `8.36` which doesn't have the mirror flag forced, so you can use any version of the EC and coreboot won't intervene.
Yes I was running 8.34 and 1.05 for sure

**Comment by divico** [comment]
> > Then start the EFI shell, and it works. I have a USB stick with an activity LED, and I can clearly see that there is no activity on the USB if it is connected before turning on.
> 
> Which port? And when you follow the instructions (i.e. have it connected before turning on), does the USB appear int he EFI Shell?

I noticed the issue when using the top right USB-A port, but I haven't tried with the other ports. No the USB stick would no show up in the boot manager list unless it is removed and inserted again, then it would appear.
This was happening as I had the battery at 2% and the issue that the battery would not charge. In the meantime I downgraded to EC '1.01', while still on '8.34' and I could not reproduce it, i.e. it works as described in your documentation and I tested all 3 USB A ports. I rolled back to the versions '8.34' / '1.05' (those installed when I had the issue) but I could not reproduce the issue neither. Now I upgraded to `8.36` and `1.10` and it also works as intended. Hence, at the moment I'm unable to reproduce the issue, no matter which versions.  Since I had the charging issue at that time, could it be that the EC was in a weird state and would somehow not detect/enable the port when turned on? I don't know.

**Comment by divico** [comment]
> Ah, it finally happened to me - it's LEARN running awry, which explains why it works for days and then doesn't. Fixed that in `1.10`.

Running coreboot `8.36`and EC `1.10` does not fix the issue for me. With only the DC jack connected, the battery full (LED purple, 63% with max charge set to 60% in coreboot), it will disconnected and reconnect randomly (at least it feels random).

Interestingly, I can systematically reproduce the following:
1. DC jack connected (only), LED purple, battery 63% with max charge set to 60%
2. Turn on laptop, stop at GDM login screen. The battery symbol says correctly that power is connected.
3. Login into a user session. Right after the session has opened, the battery symbol says (wrongly) power is disconnected. Wait ~2 seconds and observe how the symbol switches to power connected.

From there on it will regularly disconnected and reconnect.

https://user-images.githubusercontent.com/11789268/230202268-5056239f-a5dc-463b-919b-f1bdda3dedee.mp4


A few outputs, hope it helps.

```
cat /sys/class/dmi/id/ec_firmware_release
1.10
```
```
sudo ./ectool -d
EC RAM:

00: 01 0a 00 00 01 00 00 00 00 bb dd 00 00 00 00 01 
10: 00 00 00 00 00 3e 2b 00 00 00 aa aa 00 00 00 00 
20: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
30: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
40: 00 31 36 3a 35 30 3a 33 33 00 00 32 30 32 33 2f 
50: 30 34 2f 30 34 00 00 00 00 00 00 00 00 00 00 00 
60: 00 00 00 00 00 00 02 00 00 00 00 00 00 00 00 00 
70: 46 00 00 00 00 00 00 00 01 02 04 00 00 00 00 01 
80: 47 01 24 00 fc 15 1e 2d ea 15 00 00 02 00 00 f9 
90: 0d 07 2f 40 00 00 00 00 00 00 00 00 00 00 00 00 
a0: 00 00 00 00 33 90 14 00 00 00 00 00 00 00 00 00 
b0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
c0: 00 01 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
d0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
e0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
f0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
```
```
sudo nvramtool -a
boot_option = Fallback
reboot_counter = 0x0
debug_level = Debug
hyper_threading = Enable
vtd = Enable
power_profile = Balanced
me_state = Disable
me_state_counter = 0x0
wireless = Enable
webcam = Enable
microphone = Enable
legacy_8254_timer = Disable
pci_hot_plug = Disable
kbl_timeout = 30 seconds
fn_ctrl_swap = Disable
max_charge = 60%
fan_mode = Quiet
mirror_flag = Disable
mirror_flag_counter = 0x0
power_on_after_fail = Disable
fn_lock_state = 0x1
trackpad_state = 0x0
kbl_brightness = 0x2
kbl_state = 0x1
```
```
uname -a
Linux starbook 6.2.9-arch1-1 #1 SMP PREEMPT_DYNAMIC Thu, 30 Mar 2023 14:51:14 +0000 x86_64 GNU/Linux
```

**Comment by Sean-StarLabs** [comment]
```
80: 47 01 24 00 fc 15 1e 2d ea 15 00 00 02 00 00 f9 
       ^                                                              ^
   Charger is connected                     Charger is supplying 512MA
```

That looks good to me. The low current is Hybrid power; allows the laptop to draw from both the charger and battery if it needs to (the battery responds quicker).

Ignoring the icon - all behaving well?

**Comment by divico** [comment]
> ```
> 80: 47 01 24 00 fc 15 1e 2d ea 15 00 00 02 00 00 f9 
>        ^                                                              ^
>    Charger is connected                     Charger is supplying 512MA
> ```
> 
> That looks good to me. The low current is Hybrid power; allows the laptop to draw from both the charger and battery if it needs to (the battery responds quicker).
> 
> Ignoring the icon - all behaving well?

no unfortunately it is not only the icon. I believe it really disconnects the power for a moment. This becomes clearly visible when replacing the DC jack with the Starlabs USB-C 12 port hub. It will completely disconnect all attached devices, including screen randomly. I tried another dump right after it happened, maybe you can see something

```
sudo ./ectool -d
EC RAM:

00: 01 0a 00 00 01 00 00 00 00 bb dd 00 00 00 00 01 
10: 00 00 00 00 00 3e 28 00 00 00 aa aa 00 00 00 00 
20: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
30: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
40: 00 31 36 3a 35 30 3a 33 33 00 00 32 30 32 33 2f 
50: 30 34 2f 30 34 00 00 00 00 00 00 00 00 00 00 00 
60: 00 00 00 00 00 00 02 00 00 00 00 00 00 00 00 00 
70: 46 00 00 00 16 01 00 00 01 02 04 00 00 00 00 01 
80: 47 01 24 00 fc 15 1e 2d dc 15 00 00 02 00 00 83 
90: 0c 2c 2e 39 00 00 00 00 00 00 00 00 00 00 00 00 
a0: 00 00 0a 00 33 90 14 00 00 00 00 00 00 00 00 00 
b0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
c0: 00 01 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
d0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
e0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
f0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
```
In the meantime I'll go back to EC `1.01` which does not have the issue. I can do some more testing tomorrow.

**Comment by Sean-StarLabs** [comment]
Thanks, but typically - that looks happy.

I'll try and replicate it, but what I think I'll need is an EC dump when it's stopped; the EC is really quick, so either:

Leave this running to catch the disconnected
`while :; do sudo ./ectool -d; sleep 1; done > log.txt`

Or I can put together a debug build that'll capture any possible reasons (might take a few tries to narrow it down)

**Comment by david-mohr** [comment]
Here's a two minute capture with multiple disconnect events using your while loop. Hope it helps!
[log2.txt](https://github.com/StarLabsLtd/firmware/files/11165851/log2.txt)

PS I found it helped to put the system under a little load to trigger the disconnect more rapidly

**Comment by divico** [comment]
I also managed to the run the dump loop while experiencing multiple disconnects.
This is running `8.36` and EC `1.10`, LED purple, battery 56% with max charge set to 60%, Starlabs USB-C 12 Port hub connected with some USB devices and 1 screen.

[log.txt](https://github.com/StarLabsLtd/firmware/files/11167397/log.txt)

In the last seconds of the logfile I have disconnected the USB-C hub.

**Comment by Sean-StarLabs** [comment]
Thank you - same thing happening on both, `1.11` will handle that

**Comment by david-mohr** [comment]
Unfortunately, even with 1.11, I'm still getting the same problem, except it's now happening before it reaches 100%. The following log was captured at about 88%. LED is red when charging
[log4.txt](https://github.com/StarLabsLtd/firmware/files/11175112/log4.txt)

**Comment by divico** [comment]
I can confirm `1.11`does not completely fix the issue, I had several disconnects running `1.11`, but in my case they are much less frequent as compared to `1.10`and most other EC versions `>1.01`.   I managed to run the while loop while experiencing a disconnect. In first seconds of the log the laptop is running on battery (3% left, max charge still set to 60%)) with nothing connected, then I connect the USB-C 12 Port with power/screen/other USB stuff, then there is a disconnect. Hope it helps 


[log.txt](https://github.com/StarLabsLtd/firmware/files/11193008/log.txt)

**Comment by Sean-StarLabs** [comment]
@david-mohr That's memory overflow; meant it only had `0xa0` which isn't enough power - will look into it

@divico I'd expect that, new ECs are set to prioritise charging to 5% over anything else; happy to make that configurable though.

**Comment by Sean-StarLabs** [comment]
I couldn't replicate it, so I've put together `1.12` that _should_ avoid it. If it doesn't, it'll store the exact reason for stoping charging at 0xa0. i.e. in the output of `ectool`, 
EC RAM:

```
00: 01 0c 00 00 01 00 00 00 00 cc 00 00 00 00 00 01 
10: 00 00 00 00 00 3f 2a 00 00 dd 00 aa 00 00 00 00 
20: 00 00 00 00 00 00 00 00 00 00 00 00 ea 00 00 00 
30: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
40: 00 20 39 3a 35 37 3a 32 31 00 00 32 30 32 33 2f 
50: 30 34 2f 30 34 00 00 00 00 00 00 00 00 00 00 00 
60: 00 00 00 00 00 00 02 00 00 00 00 00 00 00 00 00 
70: 41 00 00 00 00 00 00 00 01 02 04 00 00 00 00 01 
80: 47 01 d9 00 fc 15 1e 2d 7d 16 00 00 90 60 01 15 
90: 16 96 33 62 00 00 00 00 00 00 00 00 00 00 00 00 
a0: **07** 00 33 90 0b 80 0a c0 14 00 01 00 00 00 00 00 
b0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
c0: 00 01 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
d0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
e0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
f0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
```
That number means:

1 = Battery Alarm Bit
2 = Battery reports fully charged
3 = Charged to target level
4 = Overcharge Protection active
5 = Battery is over temperature
6 = Slow Charging due to cold battery
7 = Normal

**Comment by david-mohr** [comment]
@Sean-StarLabs My AU adapter broke over the weekend (removing a plug next to it and the US-to-AU clip split in half), so there's a chance that my adapter was causing some of the problems. I'll get this installed today

**Comment by divico** [comment]
> @divico I'd expect that, new ECs are set to prioritise charging to 5% over anything else; happy to make that configurable though.

Prioritization makes sense, but I'm still not sure that the behavior I observe when battery <5% is really expected. Even with only the Jack connected, the power would disconnect multiple times until it reaches 5%. Above 5% the system has been running well for several days using different charger and USB-C hubs. 

If you feel like looking into the disconnects I have when the battery is <5%, I've attached two log files, one running ITE `1.11` and another running `1.12`. In both cases I have coreboot `8.36` and only the power Jack and the Starlabs GAN charger connected. I forgot to check the LEDs though.

[log ITE 1_11.txt](https://github.com/StarLabsLtd/firmware/files/11265485/log.ITE.1_11.txt)
[log ITE 1_12.txt](https://github.com/StarLabsLtd/firmware/files/11265486/log.ITE.1_12.txt)

**Comment by Sean-StarLabs** [comment]
When it's less than 5%, it's at 0.5C, which is 30W. As it's drawing no power from the battery, the second the power draw crosses 30W, the crosspoint switch will reset (you'll see it as the charger disconnecting, or if it's a USB-C display, that disconnecting) as it's an auxiliary device.

**Comment by divico** [comment]
> When it's less than 5%, it's at 0.5C, which is 30W. As it's drawing no power from the battery, the second the power draw crosses 30W, the crosspoint switch will reset (you'll see it as the charger disconnecting, or if it's a USB-C display, that disconnecting) as it's an auxiliary device.

Ok, I sort of understand thanks. I can live with that, it is not a big deal, but it remains something I have never seen with another laptop. I fear this may confuse many other users which don't know about this detail. When you are low on bat and you connect the charger you pay special attention to check if the laptop is indeed charging, specially when you are traveling and you are not sure that the power socket is really working (train, hostel, etc..). Hence, these disconnects will not go unnoticed in many cases. Even worse is that most modern desktop env will give you a warning that the machine will soon hibernate. So you just plugged the charger and few minutes / seconds later the machine tells you it must hibernate... Unless you know the technical detail about the crosspoint switch  reset, you will most likely think that there is something wrong with the power socket, the charger or the cable and you will spend time troubleshoot a problem that doesn't exist. Just my 2 cents from a user perspective.

 Beside this minor thing, in my case the laptop works very well now. Big thank you for solving the various charging issues over the last weeks so quickly.

**Comment by Sean-StarLabs** [comment]
It's a rock and a hard place; most laptops would control the power consumption based on available power - that can't be done in coreboot.

Before, we hid the actual status of the battery, and you only really used 90% of it; the top and bottom 5% were reserved to avoid this and overcharging the battery.

Hundreds of requests (originating from concerns about overcharging and calibration running slightly astray) has made it change so you now get full visibility (and a splash more runtime).

I need to re-write all the code that controls the crosspoint anyway (so it's open-source), so I'll see if there's anything we can do when doing that.

**Comment by david-mohr** [comment]
Seems to be working well for me too. Thanks for iterating with us so quickly!

**Comment by david-mohr** [comment]
We've had another laptop (running 1.12) that's stopped charging. It won't take a charge via AC or USB and the charge state is below 20% so can't update the bios (8.25). What can we do from here to get it charging again?

**Comment by Sean-StarLabs** [comment]
The BIOS can't stop it charging, but you'll be able to update the EC via the EFI Shell to `1.13`. If that doesn't work, please ping the details to support@starlabs.systems

**Comment by david-mohr** [comment]
Thanks Sean,

After some more experimenting, it look like a faulty connection between the charger and the laptop. Not quite sure which bit is at fault, but we'll run some more tests next week when we have a couple of the same laptops in the same room. But at this stage, not a software issue.

Sorry for the false alarm!
